backend/app/controllers/spree/admin/products_controller.rb

```diff
@@ -30,6 +30,11 @@ def update
             param_attrs[:option_value_ids] = param_attrs[:option_value_ids].split(',')
           end
         end
+        if updating_variant_image_rules?
```

Comment:
i think a separate nested controller might make sense here rather than shoehorning into the current one. Probably same goes for variant property rules. Not sure if this is the right time to make that change or not.